pmlc/dialect/stripe/rewrites.cc

```diff
@@ -0,0 +1,186 @@
+}
+
+void RemoveRangeOneIndexes::rewrite(ParallelForOp op, mlir::PatternRewriter& rewriter) const {
+  llvm::SmallVector<int64_t, 8> ranges;
+  auto zero = rewriter.create<AffinePolyOp>(op.getLoc(), AffinePolynomial());
+  auto body = &op.inner().front();
+  for (size_t i = 0; i < op.ranges().size(); i++) {
+    if (op.getRange(i) == 1) {
+      body->getArgument(i)->replaceAllUsesWith(zero);
+    } else {
+      ranges.push_back(op.getRange(i));
+    }
+  }
+  auto rop = rewriter.create<ParallelForOp>(op.getLoc(), ranges);
+  auto rbody = &rop.inner().front();
+  size_t new_id = 0;
+  for (size_t i = 0; i < op.ranges().size(); i++) {
+    if (op.getRange(i) != 1) {
+      body->getArgument(i)->replaceAllUsesWith(rbody->getArgument(new_id++));
+    }
+  }
+  rbody->getOperations().splice(std::prev(rbody->end(), 1), body->getOperations(), body->begin(),
+                                std::prev(body->end(), 1));
+  rewriter.eraseOp(op);
+}
+
+mlir::PatternMatchResult RemoveTrivialConstraints::matchAndRewrite(ConstraintOp op,
+                                                                   mlir::PatternRewriter& rewriter) const {
+  auto irange = AffineRange(AffinePolynomial(op.input()));
+  auto oblock = op.getOperation()->getBlock();
+  if (irange.min >= 0) {
+    // Always true
+    if (!op.ge_case().empty()) {
+      auto iblock = &op.ge_case().front();
+      oblock->getOperations().splice(Block::iterator(op), iblock->getOperations(), iblock->begin(),
+                                     std::prev(iblock->end(), 1));
+    }
+    rewriter.eraseOp(op);
+    return matchSuccess();
+  } else if (irange.max < 0) {
+    // Always false
+    if (!op.lt_case().empty()) {
+      auto iblock = &op.lt_case().front();
+      oblock->getOperations().splice(Block::iterator(op), iblock->getOperations(), iblock->begin(),
+                                     std::prev(iblock->end(), 1));
+    }
+    rewriter.eraseOp(op);
+    return matchSuccess();
+  } else {
+    // Not trivial, never mind
+    return matchFailure();
+  }
+}
+
+mlir::PatternMatchResult SplitParallelFor::matchAndRewrite(ParallelForOp pf, mlir::PatternRewriter& rewriter) const {
+  if (!SafeConstraintInterior(pf)) {
+    return matchFailure();
+  }
+  // Extract the constraint op + poly
+  auto pf_block = &pf.inner().front();
+  auto con = mlir::dyn_cast<ConstraintOp>(*std::prev(pf_block->end(), 2));
+  auto poly = AffinePolynomial(con.input());
+  if (poly.terms.size() != 1) {
+    return matchFailure();  // Only works for simple polynomials
+  }
+  // Get the actual relevant numbers from the poly
+  mlir::BlockArgument* ba = poly.terms.begin()->first;
+  // Find which argument it is (or not in this PF)
+  int which_arg = -1;
+  for (size_t i = 0; i < pf_block->getNumArguments(); i++) {
+    if (pf_block->getArgument(i) == ba) {
+      which_arg = i;
+      break;
+    }
+  }
+  if (which_arg < 0) {
+    return matchFailure();
+  }
+  int64_t m = poly.terms.begin()->second;
```

Comment:
Single variable names are hard to track and make later changes to.

Reply:
This is a case of mathematical convention.